google/validation.go

```diff
@@ -14,7 +14,7 @@ import (
 
 const (
 	// Copied from the official Google Cloud auto-generated client.
-	ProjectRegex    = "(?:(?:[-a-z0-9]{1,63}\\.)*(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?):)?(?:[0-9]{1,19}|(?:[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?))"
+	ProjectRegex    = "(?:(?:[-a-z0-9]{1,63}\\.)*(?:[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?):)?(?:[0-9]{1,19}|(?:[a-z0-9](?:[-a-z0-9]{0,61}[a-z0-9])?)|-)"
```

Comment:
yeah, the the `-` is the default (current context) for the project:
```projects/-/serviceAccounts/{ACCOUNT_EMAIL_OR_UNIQUEID}```

i can make a new regex for just that, seeing if that regex is used for the non-shorthand current context

see
-  [generateAccessToken90](https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/generateAccessToken)

Reply:
Gotcha. However I'm not seeing how this is used right now. You aren't validating the project id in this data source so I don't the point of adding this.

Reply:
ok, now i remember why i added that in:
 the `delegates` parameter takes the fully qulified name:

```terraform
data "google_impersonated_credential" "default" {
 provider = "google"
 target_service_account = "impersonated-account@fabled-ray-104117.iam.gserviceaccount.com"
 delegates = [ "projects/-/serviceAccounts/svc-2-429@mineral-minutia-820.iam.gserviceaccount.com" ]
 scopes = ["userinfo-email", "cloud-platform"]
 lifetime = "300s"
}
```

the `-` can also be the long form project name.  I clearly see that adding a regex exception shortform (`-`) into `ProjectRegex` isn't right since `ProjectRegex` can be used anywhere else outside of the shortform context.   

so...i'm trying to think of a clean way i can use `ProjectRegex` and also append `-` just for my usecase so far....i could just copy `ProjectRegex` to `ProjectRegexShort` or something and use that as

```
ServiceAccountLinkRegexPrefix = "projects/" + ProjectRegexShort + "/serviceAccounts/"
```

(i want to use the regex thats there already ...string append doesn't seem right.  a copy is just ok...or i coudl spend a lot of time to create the regex via api calls..... any other thoughts?  just the shortname's ok?

Reply:
ok, i just copied and added in `ProjectRegexWildCard` and `ServiceAccountLinkRegexPrefix = "projects/" + ProjectRegexWildCard + "/serviceAccounts/"`

Reply:
pls reopen or mark unresolved if you'd prefer some other approach to validate